python/cuml/test/test_random_forest.py

```diff
@@ -730,3 +731,63 @@ def test_rf_host_memory_leak(large_clf, estimator_type):
     # Some tiny allocations may occur, but we shuld not leak
     # without bounds, which previously happened
     assert (final_mem - initial_baseline_mem) < 2e6
+
+
+@pytest.mark.memleak
+@pytest.mark.parametrize('estimator_type', ['regression', 'classification'])
+def test_concat_memory_leak(large_clf, estimator_type):
```

Comment:
Not important for this PR, but might also be useful to pickle the resulting concatenated RF in a loop.

Reply:
Ah, yes, that would be nice too... I think should factor out the host memory leak test to be something repeatable. Maybe a context manager like:

```
with bounded_memory_usage(1e3):
   # do some stuff
```
 that will assert if memory usage grows by the end